chunk/inmemory_storage_client.go

```diff
@@ -125,7 +125,8 @@ func (m *MockStorage) BatchWrite(_ context.Context, batch WriteBatch) error {
 			items = append(items, mockItem{})
 			copy(items[i+1:], items[i:])
 		} else {
-			return fmt.Errorf("Dupe write")
+			// Ignore duplicate write
+			continue
```

Comment:
One solution would be to move the metric name entry from `GetWriteEntries` to something like `GetMetricNameWriteEntries` and then only call it in `calculateDynamoWrites` if we haven't seen the metric name before.